pkg/adaptor/elasticsearch/clients/registry.go

```diff
@@ -0,0 +1,36 @@
+package clients
+
+import (
+	"net/http"
+	"net/url"
+	"sync"
+
+	"github.com/compose/transporter/pkg/client"
+	"github.com/hashicorp/go-version"
+)
+
+// VersionedClient encapsulates a version.Constraints and Creator func that can be stopred in
+// the Clients map.
+type VersionedClient struct {
+	Constraint version.Constraints
+	Creator    Creator
+}
+
+// Creator defines the func signature expected for any implementing client.Writer
+type Creator func(chan struct{}, *sync.WaitGroup, *ClientOptions) (client.Writer, error)
+
+// Clients contains the map of versioned clients
+var Clients = map[string]*VersionedClient{}
```

Comment:
planning on addressing this in https://github.com/compose/transporter/issues/234